Refactor intance files single file

Tuzosdaniel12 opened the pull request:
From @Tuzosdaniel12 

- Refactor files of creating controllers instance into single file.
- Updated routes files to require controllers from index file in Controllers folder.
- Changed they way io.js access roomCtrl by using library.js

From @ac524 

**Addressed inconsistency with feed entry ctrl exports.**
- Created new searchCtrls utility function to search for ctrls by partial path matches
- Refactored routes/feeds to use new utility function instead of the nested object collection of ctrls.

ac524 reviewed (approved):
Changes looks good @Tuzosdaniel12. Just add a bit to the pull request description for your initial changes this is merged.

I built in one new `searchCtrls` utility function to allow us search for ctrls in the library by partial key paths. This allowed me to replace the need for the nested object syntax being used to collect that set of ctrls.

**Example**
`searchCtrls("feed.item")` returns a Map set of ctrls containing the `feed.item.comment`, `feed.item.elevate`, and `feed.item.deelevate` ctrls.

Tuzosdaniel12 commented:
•Refactor files of creating controllers instance into single file.
•Updated routes files to require controllers from index file in Controllers folder.
•Changed they way io.js access roomCtrl by using library.js
